File: src/Jering.DynamicForms/DynamicControlValidatorData.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Reflection;

namespace Jering.DynamicForms
{
    /// <summary>
    /// Holds data that defines a control validator
    /// </summary>
    public class DynamicControlValidatorData
    {
        /// <summary>
        /// Validator name  
        /// </summary>
        public string Name { get; set; }
        /// <summary>
        /// Error message
        /// </summary>
        public string ErrorMessage { get; set; }
        /// <summary>
        /// Validator specific options
        /// </summary>
        public Dictionary<string, string> Options { get; set; }
    }
}

File: src/jering-vectorartkit-webclient/src/vak-app/shared/dynamic-forms/dynamic-control/dynamic-control-validator-data.ts
﻿/**
 * Data used to defined a DynamicControlValidator
 */
export class DynamicControlValidatorData {
    name: string;
    errorMessage: string;
    options: { [key: string]: string };

    constructor(options: {
        name: string,
        errorMessage: string,
        options: { [key: string]: string }
    }) {
        this.name = options.name;
        this.errorMessage = options.errorMessage;
        this.options = options.options;
    }
}